File: ipification-sdk-react-native-js/ios/RNCoverageService.m
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(RNCoverageService,NSObject)
RCT_EXTERN_METHOD(checkCoverage: (RCTResponseSenderBlock)callbackSuccess failed: (RCTResponseSenderBlock)callbackError)
@end

File: ipification-sdk-react-native-js/ios/RNCoverageService.swift
import Foundation
import IPificationSDK

@objc(RNCoverageService) class RNCoverageService: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {return true}
  
  
  @objc func checkCoverage(_ callbackSuccess:  @escaping RCTResponseSenderBlock,failed callbackError: @escaping RCTResponseSenderBlock) {

   do {
       let coverageService = CoverageService()
       coverageService.callbackFailed = { (error) -> Void in
//           print(error.localizedDescription)
            callbackError([NSNull(), error.localizedDescription])
       }
       coverageService.callbackSuccess = { (response) -> Void in
//           print("check coverage result: ", response.isAvailable())
            callbackSuccess([NSNull(), response.isAvailable()])
       }
       try coverageService.checkCoverage()
   } catch{
       print("Unexpected error: \(error).")
       callbackError([NSNull(), error.localizedDescription])
   }
  }
    
}
